src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -276,26 +276,53 @@ internal static class StreamHelper
 
         #region Static Methods
 
-        internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet, CancellationToken cancellationToken)
+        internal static void WriteToStream(Stream input, Stream output, WebRequestPSCmdlet cmdlet, CancellationToken cancellationToken)
         {
             if (cmdlet == null)
             {
                 throw new ArgumentNullException(nameof(cmdlet));
             }
 
-            Task copyTask = input.CopyToAsync(output, cancellationToken);
-
             ProgressRecord record = new(
                 ActivityId,
                 WebCmdletStrings.WriteRequestProgressActivity,
                 WebCmdletStrings.WriteRequestProgressStatus);
 
+            // If the client loses network connectivity for over 1 minute, the task returned from 
+            // 'CopyToAsync()' will never complete and therefore the loop below will be infinite.
+            // Short explaination:
+            // - CopyToAsync() uses ReadAsync() in loop until ReadAsync() returns 0, indicating EOF.
+            // - ReadAsync() read from network Socket.
+            // - If losing network connectivity for over 1 minute, the Socket will lose connect to target service
+            //   but OS never close the Socket (tested on Windows).
+            // - Then Socket never return anything, and thus ReadAsync() will be infinitely blocked on the Socket
+            //   and CopyToAsync() task will be never completed.
+            //
+            // Since cancelation token in CopyToAsync() applies to whole CopyToAsync()
+            // and it is not restarted for very internal ReadAsync()
+            // we have to use a workaround to cancel the ReadAsync() which is blocked on Socket longer Timeout.
+            // The workaround is to reset the cancelation timer
+            // while the length of the output file is constantly increasing.
+            long previousLength = 0;
+            var timeout = cmdlet.TimeoutSec == 0 ? Timeout.Infinite : cmdlet.TimeoutSec * 1000;
+            using var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            cts.CancelAfter(timeout);
+            Task copyTask = input.CopyToAsync(output, cts.Token);
```

Comment:
```suggestion
            Task copyTask = input.CopyToAsync(output, cts.Token);
            cts.CancelAfter(timeout);
```

Reply:
Why do you think it is important?
I follow HttpClient implementation. It activates timeout before starting an operation.
https://source.dot.net/#System.Net.Http/System/Net/Http/HttpClient.cs,185

Reply:
It's not that important. But in the `HttpClient.cs` implementation, it's using `await` and thus you cannot start the timer after calling `await base.SendAsync`, and you have to start the timer before calling `SendAsync`.